doc.go

```diff
@@ -0,0 +1,7 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// gonum is a set of packages implementing numeric and scientific algorithms
```

Comment:
I'm not wildly happy about the phrasing here, or the need for a doc.go for the root (see https://godoc.org/golang.org/x/tools). One day we might have something here - maybe a version string? - but I think this can be left empty at the moment.